cfme/physical/physical_chassis.py

```diff
@@ -0,0 +1,294 @@
+        # from the UI
+        for inventory in inventory_to_match:
+            logger.debug("Validating inventory {} of {}".format(inventory, self.name))
+            try:
+                cfme_inventory = getattr(self, inventory)(method='ui' if ui else None)
+                chass_inventory = chassis_inventory[inventory]
+
+                if chass_inventory != cfme_inventory:
+                    msg = "The {} inventory does not match. (server: {}, server inventory: {}, " \
+                          "cfme inventory: {})"
+                    raise StatsDoNotMatch(msg.format(inventory, self.name, chass_inventory,
+                                                     cfme_inventory))
+            except KeyError:
+                raise HostStatsNotContains(
+                    "Server inventory information does not contain '{}'".format(inventory))
+            except AttributeError:
+                msg = "Provider does not know how to get '{}'"
+                raise ProviderHasNoProperty(msg.format(inventory))
+
+
+@attr.s
+class PhysicalChassisCollection(BaseCollection):
+    """Collection object for the :py:class:`cfme.physical.physical_chassis.PhysicalChassis`."""
+
+    ENTITY = PhysicalChassis
+
+    def all(self, provider):
+        """
+        Return all `physical_chassis` objects of the given provider.
+
+        Args:
+            provider: the provider that the physical chassis belongs to
+        """
+        physical_chassis_table = self.appliance.db.client['physical_chassis']
+        ems_table = self.appliance.db.client['ext_management_systems']
+        physical_chassis_query = (
+            self.appliance.db.client.session
+                .query(physical_chassis_table.name, physical_chassis_table.ems_ref, ems_table.name)
+                .join(ems_table, physical_chassis_table.ems_id == ems_table.id))
+
+        if self.filters.get('provider'):
+            physical_chassis_query = physical_chassis_query.filter(ems_table.name == provider.name)
+        physical_chassiss = []
+        for name, ems_ref, ems_name in physical_chassis_query.all():
+            physical_chassiss.append(self.instantiate(name=name, ems_ref=ems_ref,
+                                    provider=provider or get_crud_by_name(ems_name)))
+        return physical_chassiss
+
+
+class ComputePhysicalInfrastructureChassisView(BaseLoggedInPage):
+    """Common parts for chassis views."""
+    title = Text('.//div[@id="center_div" or @id="main-content"]//h1')
+
+    @property
+    def in_compute_physical_infrastructure_chassis(self):
+        return (self.logged_in_as_current_user and
+                self.navigation.currently_selected == ["Compute", "Physical Infrastructure",
+                                                       "Chassis"])
+
+
+class PhysicalChassisEntity(JSBaseEntity):
+    @property
+    def data(self):
+        data_dict = super(PhysicalChassisEntity, self).data
+        if 'quadicon' in data_dict and data_dict['quadicon']:
```

Comment:
```suggestion
        if data_dict.get('quadicon', ''):
```

Reply:
Also, please decide either for single quotes or double. You're mixing them from function to function :trollface:

Reply:
👍